src/Root.ts

```diff
@@ -3,7 +3,14 @@ import { defineComponent, h } from 'vue';
 import App from './App.vue';
 import * as providerMap from './providers';
 
-const providers = Object.values(providerMap);
+// The order registration of the providers is important but Object.values behaves differently in the vite bundle.
+// Setting explicit execution order in the following array:
+const providers = [
+  providerMap.UserSettingsProvider,
+  providerMap.TokenListProvider,
+  providerMap.TokensProvider,
+  providerMap.AppProvider,
+];
```

Comment:
Why is this the case? This may cause unintended issues in some of the critical flows because pool tokens values/addresses are passed around in arrays where the order is very important.

Reply:
The old code was behaving differently in webpack vs vite cause: 
```
import * as providerMap from './providers'; // providers exports A,B,C,D
Object.values(providerMap) // this returns A,B,C,D in webpack but D,C,B,A in vite
```
I guess that this is due to differences in the transpilation of the import between webpack (using babel) and vite (using ESmodules). 

The new code is more verbose but also more explicit about the execution order. I don't  think that we are using the same pattern (import * as + Object.values) in any other place.